addons/account/models/account_invoice.py

```diff
@@ -872,6 +872,8 @@ def _refund_cleanup_lines(self, lines):
                     values[name] = line[name]
                 elif name == 'invoice_line_tax_ids':
                     values[name] = [(6, 0, line[name].ids)]
+                elif name == 'sale_line_ids':
```

Comment:
This change done for when refund invoice with option "Modify: create refund, reconcile and create a new draft invoice" then created new invoice not reached from related sale order. I think it we have to fix this issue from sale module, not from account module. I will create new issue and fix from sale module instead of account module. And will remove the change from here.